cookbooks/aws-parallelcluster-slurm/resources/mysql_client/partial/_setup_rhel_based.rb

```diff
@@ -49,6 +49,8 @@ def package_platform
     platform_version = node['platform_version'].to_i
     if platform_version == 2
       platform_version = 7
+    elsif platform_version == 2023
```

Comment:
Bad practice: redefinition of the same variable with different meaning.
Here we should have `platform_version` and `mysql_platform_version`

Reply:
Here we are using the `platform_version` is just the location in our s3 bucket where we archive the packages we want. Similar to what we do for Cinc-client

Reply:
Sure, the bad practice is doing:

```
platform_version = define platform version with meaning 1
if platform_version == 2
      platform_version = 7
use platform version with meaning 2
```

Not a blocking comment because the practice is not introduced in this PR, but better to address it at least in a follow up PR